src/ore/src/future.rs

```diff
@@ -62,6 +62,26 @@ pub trait OreFutureExt {
         Self: Future + Send + 'static,
         Self::Output: Send + 'static;
 
+    /// The same as `run_in_task`, but allows the callee to dynamically choose whether or
+    /// not the future is polled into a Tokio task.
+    async fn optionally_run_in_task<Name, NameClosure>(
+        self,
+        nc: NameClosure,
+        in_task: bool,
+    ) -> Self::Output
+    where
```

Comment:
What do you think about this slightly different name/parameter order? I feel like it reads a little better!

```suggestion
    async fn run_in_task_if<Name, NameClosure>(
        self,
        in_task: bool,
        nc: NameClosure,
    ) -> Self::Output
    where
```

Reply:
updated!